src/Controls/src/Core/NavigationPage/NavigationPageToolbar.cs

```diff
@@ -180,7 +180,11 @@ void UpdateBackButton()
 
 			// Set this before BackButtonVisible triggers an update to the handler
 			// This way all useful information is present
-			if (Parent is FlyoutPage flyout && flyout.ShouldShowToolbarButton() && !anyPagesPushed.Value)
+			if (Parent is FlyoutPage flyout && flyout.ShouldShowToolbarButton()
+#if !WINDOWS
```

Comment:
Could include a comment in code to explain why the behavior needs to be different on Windows?

Reply:
@jsuarezruiz , I have added a comment. please let me know if you have any concern.